core/jobs/batch_jobs/exp_migration_jobs.py

```diff
@@ -0,0 +1,502 @@
+            })
+            yield (exp_id, exp_change)
+
+    @staticmethod
+    def _delete_exploration_from_cache(
+        exploration: exp_domain.Exploration
+    ) -> result.Result[str, Exception]:
+        """Deletes exploration from cache.
+
+        Args:
+            exploration: Exploration. The exploration which should be deleted
+                from cache.
+
+        Returns:
+            Result(str, Exception). The ID of the skill when the deletion was
+            successful or Exception when the deletion failed.
+        """
+        try:
+            caching_services.delete_multi(
+                caching_services.CACHE_NAMESPACE_EXPLORATION,
+                None, [exploration.id])
+            return result.Ok(exploration.id)
+        except Exception as e:
+            return result.Err(e)
+
+    @staticmethod
+    def _update_exp_summary(
+        migrated_exp: exp_domain.Exploration,
+        exp_summary_model: exp_models.ExpSummaryModel,
+        exp_rights_model: exp_models.ExplorationRightsModel
+    ) -> exp_models.ExpSummaryModel:
+        """Generates a newly updated exploration summary model.
+
+        Args:
+            migrated_exp: Exploration. The migrated exploration domain object.
+            exp_summary_model: ExpSummaryModel. The exploration summary model
+                to update.
+            exp_rights_model: ExplorationRightsmodel. The exploration rights
+                model used to update the exploration summary.
+
+        Returns:
+            ExpSummaryModel. The updated exploration summary model to put into
+            the datastore.
+        """
+        exp_summary = exp_services.compute_summary_of_exploration( # type: ignore[no-untyped-call]
+            migrated_exp, exp_rights_model, exp_summary_model,
+            skip_exploration_model_last_updated=True)
+        exp_summary.version += 1
+        updated_exp_summary_model: exp_models.ExpSummaryModel = (
+            exp_services.populate_exp_summary_model_fields( # type: ignore[no-untyped-call]
+                exp_summary_model, exp_summary
+            )
+        )
+
+        return updated_exp_summary_model
+
+    @staticmethod
+    def _update_exploration(
+        exp_model: exp_models.ExplorationModel,
+        exp_rights_model: exp_models.ExplorationRightsModel,
+        migrated_exp: exp_domain.Exploration,
+        exp_changes: Sequence[exp_domain.ExplorationChange]
+    ) -> Sequence[base_models.BaseModel]:
+        """Generates newly updated exploration models.
+
+        Args:
+            exp_model: ExplorationModel. The exploration which should be
+                updated.
+            exp_rights_model: ExplorationRightsModel. The exploration rights
+                model which is to be updated.
+            migrated_exp: Exploration. The migrated exploration domain
+                object.
+            exp_changes: sequence(ExplorationChange). The exploration changes
```

Comment:
Probably should be Sequence(...) or list(...)

Reply:
Updated.